### PR TITLE
Stp attack play

### LINKED ISSUE #965: Attack Play

### PR DESCRIPTION
**Pull request by jordiluong**
Added roles and invariants to attack play. Roles can still be improved at a later stage

Closes #965

### CODE REVIEW

**Review by RobotJente — APPROVED** [review]

File: src/stp/new_plays/Attack.cpp [hunk 1]
```diff
@@ -3,65 +3,116 @@
...
     startPlayInvariants.emplace_back(std::make_unique<invariant::WeHaveBallInvariant>());
+    startPlayInvariants.emplace_back(std::make_unique<invariant::GoalVisionInvariant>());
 
-    // TODO: decide keep invariants
-/*    keepPlayInvariants.clear();
-    keepPlayInvariants.emplace_back(std::make_unique<invariant::WeHaveBallInvariant>());*/
+    keepPlayInvariants.clear();
+    keepPlayInvariants.emplace_back(std::make_unique<invariant::NormalPlayGameStateInvariant>());
+    keepPlayInvariants.emplace_back(std::make_unique<invariant::BallCloseToUsInvariant>());
+    keepPlayInvariants.emplace_back(std::make_unique<invariant::BallIsFreeInvariant>());
```
Comment (timovdk) [inline src/stp/new_plays/Attack.cpp]: So this play will only be kept if we don't have the ball? How can we shoot the ball then?

**Review by timovdk — CHANGES_REQUESTED** [review]
Looks good, but I don't think the keep invariants work with the attacker role

File: src/stp/new_plays/Attack.cpp (hunk 1 above)
Comment (jordiluong) [reply src/stp/new_plays/Attack.cpp]: Oops hehe

File: src/stp/new_plays/Attack.cpp [hunk 2]
```diff
@@ -3,65 +3,116 @@
 //
 
 #include "stp/invariants/WeHaveBallInvariant.h"
+#include "stp/invariants/GoalVisionInvariant.h"
+#include "stp/invariants/BallCloseToUsInvariant.h"
+#include "stp/invariants/BallIsFreeInvariant.h"
```
Comment (jordiluong) [inline src/stp/new_plays/Attack.cpp]:
```suggestion
```

File: src/stp/new_plays/Attack.cpp (hunk 1 above)
Comment (jordiluong) [inline src/stp/new_plays/Attack.cpp]:
```suggestion
```

**Pull request merged**